multiversx_sdk/core/transactions_factories/transfer_transactions_factory.py

```diff
@@ -76,3 +76,34 @@ def create_transaction_for_esdt_token_transfer(self,
             gas_limit=extra_gas_for_transfer,
             add_data_movement_gas=True
         ).build()
+
+    def create_transaction_for_transfer(self,
+                                        sender: IAddress,
+                                        receiver: IAddress,
+                                        native_amount: Optional[int] = None,
+                                        token_transfers: Optional[Sequence[ITokenTransfer]] = None,
+                                        data: Optional[bytes] = None) -> Transaction:
+        if not native_amount and not token_transfers:
+            raise BadUsageError("No native token amount or token transfers provided")
+
+        if token_transfers and data:
+            raise BadUsageError("Can't set data field when sending esdt tokens")
+
+        if native_amount and not token_transfers:
+            return self.create_transaction_for_native_token_transfer(
+                sender=sender,
+                receiver=receiver,
+                native_amount=native_amount,
+                data=data.decode() if data else None
+            )
+
+        token_transfers = list(token_transfers) if token_transfers else []
```

Comment:
The optional refactoring idea from above, if applicable.

Reply:
I can do it, but I am not sure it's what we want.